### ISSUE TITLE
Not working on iOS 16

### ISSUE THREAD

**Issue by fizawisedv** [umbertoghio/react-native-screen-recorder#7]
I am getting the below error on iOS 16. Recording works well on iOS 14 as I have tested.

`this._nativeModule.addListener is not a function. (In 'this._nativeModule.addListener(t)', 'this._nativeModule.addListener' is undefined)`

**Comment by umbertoghio** [comment]
Hi @fizawisedv , can you share the code snippet where you have this issue?

**Comment by fizawisedv** [comment]
It occurs at the `startRecording()` method. Below is my code.

```
    try {
        const recordMic = await PermissionService.checkPermission("mic") // ask for mic permission
        const options = {
            mic: recordMic, // defaults to true
            androidBannerStopRecordingHandler: this.androidBannerStopRecordingHandler
        }

        const recordingStatus = await ScreenRecorder.startRecording(options).catch((error) => {
            LOG.error("RECORD SCREEN ERROR: ", error)
        })

        if (recordingStatus === 'started') {
            this.isRecording = true
        }
        else if (recordingStatus === 'userDeniedPermission') SimpleToast.show('Please grant permission to record screen')
        else SimpleToast.show('Recording not supported')
    } catch(e) {
        LOG.error("Recording error: ", e) // this log shows the above error
    }
```

**Comment by umbertoghio** [comment]
thanks, will have a look soon

**Comment by umbertoghio** [comment]
Hi @fizawisedv , 

the issue seems to be caused by the androidBannerStopRecordingHandler option I added recently to improve handling on Android, however, it seems that the implementation is not consistent in iOS.

As a workaround only set androidBannerStopRecordingHandler on Android

```
const options = {
            mic: recordMic, // defaults to true
            androidBannerStopRecordingHandler: Platform.OS === 'ios' ? null : this.androidBannerStopRecordingHandler
        }
```

I'll handle this properly on the next iteration

**Comment by umbertoghio** [comment]
Fixed in 0.1.4
